integration/loader/main.go

```diff
@@ -140,62 +131,58 @@ func main() {
 		log.Fatal().Err(err).Msgf("unable to initialize Flow supervisor client")
 	}
 
-	go func() {
-		// run load cases
-		for i, c := range cases {
-			log.Info().Str("load_type", *loadTypeFlag).Int("number", i).Int("tps", c.tps).Dur("duration", c.duration).Msgf("Running load case...")
-
-			loaderMetrics.SetTPSConfigured(c.tps)
-
-			var lg *utils.ContLoadGenerator
-			if c.tps > 0 {
-				var err error
-				lg, err = utils.NewContLoadGenerator(
-					log,
-					loaderMetrics,
-					flowClient,
-					supervisorClient,
-					loadedAccessAddr,
-					priv,
-					&serviceAccountAddress,
-					&fungibleTokenAddress,
-					&flowTokenAddress,
-					c.tps,
-					*accountMultiplierFlag,
-					utils.LoadType(*loadTypeFlag),
-					*feedbackEnabled,
-					utils.ConstExecParam{
-						MaxTxSizeInByte: *maxConstExecTxSizeInBytes,
-						AuthAccountNum:  *authAccNumInConstExecTx,
-						ArgSizeInByte:   *argSizeInByteInConstExecTx,
-						PayerKeyCount:   *payerKeyCountInConstExecTx,
-					},
-				)
-				if err != nil {
-					log.Fatal().Err(err).Msgf("unable to create new cont load generator")
-				}
-
-				err = lg.Init()
-				if err != nil {
-					log.Fatal().Err(err).Msgf("unable to init loader")
-				}
-				lg.Start()
+	// run load cases
+	for i, c := range cases {
+		log.Info().Str("load_type", *loadTypeFlag).Int("number", i).Int("tps", c.tps).Dur("duration", c.duration).Msgf("Running load case...")
+
+		loaderMetrics.SetTPSConfigured(c.tps)
+
+		var lg *utils.ContLoadGenerator
+		if c.tps > 0 {
+			var err error
+			lg, err = utils.NewContLoadGenerator(
+				log,
+				loaderMetrics,
+				flowClient,
+				supervisorClient,
+				loadedAccessAddr,
+				priv,
+				&serviceAccountAddress,
+				&fungibleTokenAddress,
+				&flowTokenAddress,
+				c.tps,
+				*accountMultiplierFlag,
+				utils.LoadType(*loadTypeFlag),
+				*feedbackEnabled,
+				utils.ConstExecParam{
+					MaxTxSizeInByte: *maxConstExecTxSizeInBytes,
+					AuthAccountNum:  *authAccNumInConstExecTx,
+					ArgSizeInByte:   *argSizeInByteInConstExecTx,
+					PayerKeyCount:   *payerKeyCountInConstExecTx,
+				},
+			)
+			if err != nil {
+				log.Fatal().Err(err).Msgf("unable to create new cont load generator")
 			}
 
-			// if the duration is 0, we run this case forever
-			if c.duration.Nanoseconds() == 0 {
-				return
+			err = lg.Init()
+			if err != nil {
+				log.Fatal().Err(err).Msgf("unable to init loader")
 			}
+			lg.Start()
+		}
 
-			time.Sleep(c.duration)
-
-			if lg != nil {
-				lg.Stop()
-			}
+		// if the duration is 0, we run this case forever
+		if c.duration.Nanoseconds() == 0 {
+			return
```

Comment:
This should be an infinite sleep loop?

Reply:
Good catch, this behavior changed without the context/subroutine.